### DISCUSSION TITLE
How to use (linux) local users as fileBrowser users

### DISCUSSION THREAD

**Discussion by mmonlouis** [filebrowser/filebrowser#3368] [General]
Hello, 

I need some help to setup FileBrowser. 
I have a samba server configured.  
All shared directories belongs to users manage in FreeIPA server. Access rights are set to files.
I want to setup fileBrowser on top of this, to offer an easy to use interface to users.
I dont need filebrowser to manager files right. It just have to access to file system with (PAM) user. 
Can it be done ? or FileBrowser can only access files with his own running user.

Thanks for your help

**Comment by ColCh** [comment]
that's not possible - I just spinned up multiple instances of filebrowser

to use same filebrowser instance:

if you need to redefine UID/GID for entire filebrowser, use this

https://github.com/filebrowser/filebrowser/issues/373

if you need to define different UID for each filebrowser user, that's not possible

container starts from UID

https://github.com/filebrowser/filebrowser/blob/3d6c5152fe22a4a47783253a3506a2930b7b6be3/docker/root/etc/services.d/filebrowser/run#L3

and uses same UID when writing files

https://github.com/filebrowser/filebrowser/blob/3d6c5152fe22a4a47783253a3506a2930b7b6be3/http/resource.go#L265

this is what filebrowser uses for `fs` of particular user:

https://github.com/filebrowser/filebrowser/blob/3d6c5152fe22a4a47783253a3506a2930b7b6be3/users/users.go#L97

afero fs is this: https://github.com/spf13/afero

it doesn't support different UID/GID for fs, so in filebrowser we need to patch / workaround this.

but also keep in mind that entire process is running from particular UID/GID (see top lines of this message), so process should have access to change that
